Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -1289,7 +1289,7 @@ namespace OpenInfraPlatform {
 									size_t i = 0;
 									for (const auto& point : faceSet->Coordinates->CoordList)
 									{
-										if (i == outerLoop) {
+										if (i == outerLoop-1) {

```

Comment:
this if does not make any sense - why is it there?

Reply:
Because faceSet->Coordinates->CoordList is a vector of coordinates, which, as we know, goes always from 0 to n-1. 
However, indices here are from 1 to n. Thus, to select correct coordinates of the point we should either subtract 1 from indices, or add 1 to i in this if statement

Reply:
Does it work, though? I have a feeling that this is wrong. Please also observe how the indices are handled above, with no voids in the face.

Reply:
With: 
![image](https://user-images.githubusercontent.com/62873079/118398049-330f9000-b657-11eb-8469-10705494f1cd.png)

Without: 
![image](https://user-images.githubusercontent.com/62873079/118398054-3efb5200-b657-11eb-86a0-f14d99564cc9.png)

As you can see, if I do not use -1,  a list of inner points have size of 3  (should have 4) + outer points are not on the one plane

Reply:
interesting ... :) well, as long as it works, I don't mind any solution.